Nitaicandra/CENPROJECT
# Booking backend

## Summary of changes
- added functionality for customers to create a booking
- systems checks for conflicts automatically and creates booking if possible
- added functionality to retrieve upcoming and past bookings for a logged-in user

## How to test
POST
- Make sure you are logged in to a customer account 
- In postman, add a new request using the path `http://localhost:3001/api/bookings/book-service/serviceId`. Replace serviceId with the id for the service you want to book
- Add the customer token to the authorization tab
- In the body of the request, create a new booking following this format:

```
{
        "date": "2024-07-23",
        "weekDay": "Monday",
        "startTime": "09:00", 
        "endTime": "11:00"
}
```


GET
- Make sure you are logged in to a customer or business account 
- The paths for the requests in postman  are `http://localhost:3001/api/bookings/future` and `http://localhost:3001/api/bookings/past`
- Add the customer/business token to the authorization tab
